src/Ocelot/Authorization/ScopesAuthorizer.cs

```diff
@@ -28,7 +31,14 @@ public Response<bool> Authorize(ClaimsPrincipal claimsPrincipal, List<string> ro
             return new ErrorResponse<bool>(values.Errors);
         }
 
-        var userScopes = values.Data;
+        IList<string> userScopes = values.Data;
```

Comment:
This change isn't necessary; using `var` is fine.
The problem is that you are casting to a more abstract type, which can slightly affect performance.
What was your reason for casting the `List<string>` object?

Reply:
Arrays and Lists share the IList implementation so I can use the same pointer for both the list from the initial value and then if space-sapareted array from the `string.Split`. There is no boxing there. Later the instance will be used by the LINQ anyway. If I use `var` then I need to call `ToList` on the `string.Split` value which will be (slightly) worse performance.